lib/proxy.js

```diff
@@ -696,6 +701,24 @@ Proxy.prototype._onRequestData = function(ctx, chunk, callback) {
   });
 };
 
+Proxy.prototype._onRequestEnd = function(ctx, chunk, callback) {
```

Comment:
It's an exact copy of `_onRequestData` with a different set of event handlers. Maybe there is a way to avoid duplicated code?